src/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/process/inspectors/TestGlobalPatchAnalysis.java

```diff
@@ -41,156 +43,162 @@
+		RepairnatorConfig.getInstance().setPatchRankingMode("OVERFITTING");
+
+		cloneStep.addNextStep(new CheckoutBuggyBuild(inspector, true)).addNextStep(new TestProject(inspector))
+				.addNextStep(new GatherTestInformation(inspector, true, new BuildShouldFail(), false))
+				.addNextStep(new ComputeClasspath(inspector, true))
+				.addNextStep(new ComputeSourceDir(inspector, true, false)).addNextStep(nopolRepair)
+				.addNextStep(npeRepair);
+		cloneStep.execute();
+
+		List<RepairPatch> rankedPatchesP4J = inspector.getJobStatus().getRankedPatches(Features.P4J);
+		List<RepairPatch> rankedPatchesS4R = inspector.getJobStatus().getRankedPatches(Features.S4R);
+
+		// test ranking by P4J overfitting-scores
+		assertThat(rankedPatchesP4J.get(0).getToolname(), is("NopolSingleTest"));
+		assertEquals(rankedPatchesP4J.get(0).getOverfittingScore(Features.P4J), -2590, 1);
+		assertThat(rankedPatchesP4J.get(4).getToolname(), is("NopolSingleTest"));
+		assertEquals(rankedPatchesP4J.get(4).getOverfittingScore(Features.P4J), -410, 1);
+
+		// test ranking by S4R overfitting-scores
+		assertThat(rankedPatchesS4R.get(0).getToolname(), is("NopolSingleTest"));
+		assertEquals(rankedPatchesS4R.get(0).getOverfittingScore(Features.S4R), -0.06, 0.001);
+		assertThat(rankedPatchesS4R.get(1).getToolname(), is("NopolSingleTest"));
+		assertEquals(rankedPatchesS4R.get(1).getOverfittingScore(Features.S4R), -0.05, 0.001);
+	}
+
+	
+	@Test
+	public void testODSPatchClassification() throws IOException {
+		long buildId = 203797975; // surli/failingProject build
+		Build build = this.checkBuildAndReturn(buildId, false);
+
+		tmpDir = Files.createTempDirectory("patch_classification").toFile();
+		BuildToBeInspected toBeInspected = new BuildToBeInspected(build, null, ScannedBuildStatus.ONLY_FAIL, "");
+		ProjectInspector inspector = new ProjectInspector(toBeInspected, tmpDir.getAbsolutePath(), null, null);
+
+		CloneRepository cloneStep = new CloneRepository(inspector);
+		NopolSingleTestRepair nopolRepair = new NopolSingleTestRepair();
+		nopolRepair.setProjectInspector(inspector);
+
+		RepairnatorConfig.getInstance().setRepairTools(new HashSet<>(Arrays.asList(nopolRepair.getRepairToolName())));
+
+		cloneStep.addNextStep(new CheckoutBuggyBuild(inspector, true)).addNextStep(new TestProject(inspector))
+				.addNextStep(new GatherTestInformation(inspector, true, new BuildShouldFail(), false))
+				.addNextStep(new ComputeClasspath(inspector, true))
+				.addNextStep(new ComputeSourceDir(inspector, true, false)).addNextStep(nopolRepair);
+		cloneStep.execute();
+
+		List<RepairPatch> classifyPatcheswithODS = inspector.getJobStatus().getCorrectbessLabeledPatches(Features.ODS,
+				buildId);
+
+		// There are 10 patches are generated for this failing build
+		assertThat(classifyPatcheswithODS.size(), is(10));
+
+		int overfittingCount = 0;
+
+		for(RepairPatch patch : classifyPatcheswithODS) {
+			if ("overfitting".equals(patch.getODSLabel())){
```

Comment:
[minor] We can use an enumerator with all the possible values that ODS can generate. 
Actually, we can create one enum from this tree fields: https://github.com/SpoonLabs/coming/blob/ef0451b97086d803bf954c539b726ca6e0f82d4e/src/main/java/fr/inria/coming/codefeatures/RepairnatorFeatures.java#L36

Reply:
Hi Matias,

I thought about the enumerator before, then I realized it may be less necessary for that. As in our code, we did not compare any type of ods labels in the logic, we simply return the string type label according to probability score. WDYT?

Reply:
Hi Matias, I thought I leave comments but not succeeded :) 
I actually thought about add an enumerator of the ODS labels, then after analysis, I think it may not be necessary.  In our code, we do not have the logic comparison or other usage of the labels, we simply return the string value according to the probability score, and this value will be set as a property of the repair patch. Creating an enumerator, we will have still eventually return a string. WDYT?

Reply:
> we simply return the string value according to the probability score, and this value will be set as a property of the repair patch. Creating an enumerator, we will have still eventually return a string. WDYT?

We can return the enumerator's values e.g.

```
enum PATCH_LABEL{
   OVERFITTING, CORRECT, UNKNOWN, ....;
}
```

then the method returns a type `PATCH_LABEL`  such as (Pseudo-code):

```
public  PATCH_LABEL getLabel(patch){
    
        if(ODS.isOverfitting(patch)) 
         return  PATCH_LABEL.OVERFITTING;
        else
       return  PATCH_LABEL.CORRECT;

}

```

this would avoids us to then compared the output via strings (which can be error proud -typos in the string, case sensitive, etc-)